apps/server/src/lib/ccr.ts

```
@@ -0,0 +1,156 @@
+
+// Cache for CCR status to avoid spamming the process list
+let statusCache: { status: CCRStatus; timestamp: number } | null = null;
+const STATUS_CACHE_TTL = 2000; // 2 seconds
+
+/**
+ * Strips ANSI escape codes from a string
+ */
+function stripAnsi(str: string): string {
+  // eslint-disable-next-line no-control-regex
+  return str.replace(
+    /[\u001b\u009b][[()#;?]*(?:[0-9]{1,4}(?:;[0-9]{0,4})*)?[0-9A-ORZcf-nqry=><]/g,
```

Biome (2.3.13)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)